### PR TITLE
Save action should not move user to display of the same page

### PR DESCRIPTION
**Pull request by benwbrum**

### CODE REVIEW

**Comment by saracarl** [comment]
this woks in the simple path, but in the path where you have to categorize a new subject it kicks you back to the overview of the screen.

**Comment by saracarl** [comment]
This code breaks a number of tests:

Finished in 3 minutes 7.1 seconds (files took 2.16 seconds to load)
137 examples, 14 failures

Failed examples:

rspec ./spec/features/archive_import_spec.rb:60 # IA import actions tests ocr correction
rspec ./spec/features/collection_spec.rb:38 # collection settings js tasks checks that an added user can edit a work in the collection
rspec ./spec/features/document_sets_spec.rb:238 # document sets checks document set subject tabs
rspec ./spec/features/document_sets_spec.rb:317 # document sets checks document set breadcrumbs - page level
rspec ./spec/features/editor_actions_spec.rb:97 # editor actions transcribes a page
rspec ./spec/features/guest_spec.rb:29 # guest user actions tests guest account transcription
rspec ./spec/features/linking_spec.rb:47 # subject linking deletes a subject
rspec ./spec/features/linking_spec.rb:73 # subject linking links a categorized subject
rspec ./spec/features/linking_spec.rb:172 # subject linking links subjects on a translation
rspec ./spec/features/needs_review_spec.rb:67 # needs review marks pages as needing review
rspec ./spec/features/needs_review_spec.rb:91 # needs review marks translated pages as needing review
rspec ./spec/features/needs_review_spec.rb:103 # needs review filters list of review pages
rspec ./spec/features/needs_review_spec.rb:198 # needs review marks pages as no longer needing review
rspec ./spec/features/needs_review_spec.rb:215 # needs review marks translated pages as no longer needing review

File: spec/features/guest_spec.rb [hunk 1]
```diff
@@ -60,7 +59,7 @@
     @user = User.last
     expect(@user.login).to eq('martha') 
     expect(@guest.id).to eq(@user.id)
-    expect(page.current_path). to eq collection_translate_page_path(@collection.owner, @collection, @work, @page.id)
+    expect(page.current_path). to eq collection_transcribe_page_path(@collection.owner, @collection, @work, @page.id)
```
Comment (benwbrum) [inline spec/features/guest_spec.rb]: @tbl73 Can you review the `guest_spec` for sanity?  I believe that we should be sending guest users back to the transcribe page once they've registered, provided they were already on the transcribe page.  I'm not entirely sure how it worked at all before -- maybe one of my previous edits in this branch messed things up?

Comment (tbl73) [reply spec/features/guest_spec.rb]: I just noticed this - I'll take a look at it.  That sounds right, though.  I know generally we send users back to wherever they were before they register/log in, but it was slightly trickier with the guest users and making sure the page actually saved.

**Pull request merged**